src/gui/dialogs/new/newdialog.cpp

```diff
@@ -365,6 +385,102 @@ void NewDialog::reset_at_compile_change(bool v) {
+        ui->text_bp_bht_bits_number->setEnabled(true);
+    } else if (config->get_bp_type() == machine::PredictorType::SMITH_2_BIT) {
+        ui->select_bp_init_state->addItem(
+            "Strongly not taken", QVariant((uint8_t)machine::PredictorState::STRONGLY_NOT_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Weakly not taken", QVariant((uint8_t)machine::PredictorState::WEAKLY_NOT_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Weakly taken", QVariant((uint8_t)machine::PredictorState::WEAKLY_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Strongly taken", QVariant((uint8_t)machine::PredictorState::STRONGLY_TAKEN));
+        ui->select_bp_init_state->setCurrentIndex(
+            ((uint8_t)machine::PredictorState::WEAKLY_NOT_TAKEN) - 2);
+        config->set_bp_init_state(machine::PredictorState::WEAKLY_NOT_TAKEN);
+        ui->select_bp_init_state->setEnabled(true);
+        ui->select_bp_init_state->setEnabled(true);
+        ui->slider_bp_bhr_bits->setEnabled(true);
+        ui->text_bp_bhr_bits_number->setEnabled(true);
+        ui->slider_bp_bht_addr_bits->setEnabled(true);
+        ui->text_bp_bht_addr_bits_number->setEnabled(true);
+        ui->text_bp_bht_bits_number->setEnabled(true);
+    } else if (config->get_bp_type() == machine::PredictorType::SMITH_2_BIT_HYSTERESIS) {
+        ui->select_bp_init_state->addItem(
+            "Strongly not taken", QVariant((uint8_t)machine::PredictorState::STRONGLY_NOT_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Weakly not taken", QVariant((uint8_t)machine::PredictorState::WEAKLY_NOT_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Weakly taken", QVariant((uint8_t)machine::PredictorState::WEAKLY_TAKEN));
+        ui->select_bp_init_state->addItem(
+            "Strongly taken", QVariant((uint8_t)machine::PredictorState::STRONGLY_TAKEN));
+        ui->select_bp_init_state->setCurrentIndex(
+            ((uint8_t)machine::PredictorState::WEAKLY_NOT_TAKEN) - 2);
+        config->set_bp_init_state(machine::PredictorState::WEAKLY_NOT_TAKEN);
+        ui->select_bp_init_state->setEnabled(true);
+        ui->select_bp_init_state->setEnabled(true);
+        ui->slider_bp_bhr_bits->setEnabled(true);
+        ui->text_bp_bhr_bits_number->setEnabled(true);
+        ui->slider_bp_bht_addr_bits->setEnabled(true);
+        ui->text_bp_bht_addr_bits_number->setEnabled(true);
+        ui->text_bp_bht_bits_number->setEnabled(true);
+    } else {
+        ui->select_bp_init_state->setEnabled(false);
+        ui->slider_bp_bhr_bits->setEnabled(false);
+        ui->text_bp_bhr_bits_number->setEnabled(false);
+        ui->slider_bp_bht_addr_bits->setEnabled(false);
+        ui->text_bp_bht_addr_bits_number->setEnabled(false);
+        ui->text_bp_bht_bits_number->setEnabled(false);
+    }
+}
+
+void NewDialog::bp_init_state_change() {
+    config->set_bp_init_state(
```

Comment:
Personally, I think it would make the code less readable? Some of the functions for handling the GUI would have their own declaration, while other functions which happen to be shorter would be defined right in the connect() function call in the place of the slot. Is that correct?

If that is fine and the intended way this should be written, then I'll rewrite it, I just wanted to check.

Reply:
I mean, this is not some exposed functionality, this is purely internal to the component. And having whole clutter of function declaration and everything just to say, "if this changes, enable this" does not seems useful. I prefer the internal behavior to be in one place, not 3 (declaration, definition, connect).
I will leave it to you, but I do this a lot in other parts of the gui.

Reply:
I've made most of the slots into lambdas, as you suggested, so this should be addressed now.